xbmc/music/tags/MusicInfoTag.h

```diff
@@ -154,12 +155,22 @@ class CMusicInfoTag : public IArchivable, public ISerializable, public ISortable
    \param genre genre to add.
    */
   void AppendGenre(const std::string &genre);
+  
+  void AddArtistRole(const std::string& Role, const std::string& strArtist);
+  void AddArtistRole(const std::string& Role, const std::vector<std::string>& artists);
+  void AppendArtistRole(const CMusicRole& ArtistRole);
+  const std::string GetArtistStringForRole(const std::string& strRole) const;
+  const std::string GetContributorsText(bool bIncludeRole = false) const;
```

Comment:
avoid using bool parameters, it's very hard to know what they do without inspecting the method declaration. I would go with GetContributorsText() and GetContributorsAndRolesText() or similar